### ISSUE TITLE
GYD Constant Rate Provider

### ISSUE THREAD

**Issue by Zen-Maxi** [balancer/code-review#137]
### Type of Smart Contract

Rate Provider

### Contract Source Code

0x3a216B01db971Bf28D171C9dA44Cc8C89867697F

### Protocol Documentation

https://docs.gyro.finance/gyroscope-protocol/readme

### Protocol Audits

https://docs.gyro.finance/gyroscope-protocol/audit-reports

### Terms and Conditions

- [X] The protocol to be integrated has been audited by at least one security firm or independent researcher.
- [X] If deployed, the contract & all involved downstream contracts are verified on etherscan.
- [X] The contract does not rely upon an EOA to supply any critical data (e.g., price).
- [X] If upgradeable, the contract's administrator is not an EOA.

**Comment by Zen-Maxi** [comment]
@mkflow27 i do not think this need a thorough review as the value is fixed and has no upgradeabillity but because it is a first of its kind for a GYD gyroscope pool i requested the review. If you can take a quick glance i would say it is safe to merge to permit Gyro to proceed with their gauge request sooner than later. https://forum.balancer.fi/t/bip-xxx-enable-gauge-for-gyd-wsteth-e-clp-arbitrum/5956

**Comment by mkflow27** [comment]
@Zen-Maxi What asset will this rateProvider be used with? and what will it be paired with?
